test/ttmlir/Dialect/TTNN/eltwise/unary/cast/simple_cast.mlir

```diff
@@ -9,4 +9,25 @@ module attributes {} {
     // CHECK-SAME: tensor<64x128xbf16,
     return %1 : tensor<64x128xbf16>
   }
+
+  func.func @cast_fold(%arg0: tensor<64x128xf32>) -> tensor<64x128xf32> {
+    // CEHCK-LABEL: func.func @cast_fold
+    %0 = tensor.empty() : tensor<64x128xf32>
+    %1 = "ttir.typecast"(%arg0, %0) <{operandSegmentSizes = array<i32: 1, 1>, operand_constraints = [#any_device, #any_device]}> : (tensor<64x128xf32>, tensor<64x128xf32>) -> tensor<64x128xf32>
+    // CHECK-NOT: typecast
+    // CHECK: return %arg0 : tensor<64x128xf32
+    return %1 : tensor<64x128xf32>
+  }
+
+  func.func @cast_fold_add(%arg0: tensor<64x128xf32>, %arg1: tensor<64x128xf32>) -> tensor<64x128xf32> {
+    // CHECK-LABEL: func.func @cast_fold_add
+    // CHECK-NOT: typecast
+    // CHECK: ttnn.add
+    // CHECK-NOT: typecast
```

Comment:
Just curious, did you add this second CHECK-NOT to confirm that no typecast will be inserted by some other pass in the pipeline, and even if it gets inserted it will be folded?

Reply:
I added to second CHECK-NOT to verify that we don't have any `typecast` op in the entire graph (either before or after ttnn.add op). I see your point that it causes more confusion and some other optimization may add another typecast op in future. So I have removed the second CHECK-NOT statement.